data-tx-jdbc/src/main/java/io/micronaut/transaction/jdbc/DataSourceTransactionManager.java

```diff
@@ -247,4 +257,37 @@ protected void prepareTransactionalConnection(Connection con, TransactionDefinit
     public Connection getConnection() {
         return connectionOperations.getConnectionStatus().getConnection();
     }
+
+    /**
+     * Checks if thrown exception is from the JDBC driver telling that feature is not supported.
+     * For example, some drivers don't support {@link Connection#releaseSavepoint(Savepoint)}
+     * and we want to handle that case and continue execution.
+     *
+     * @param exception The thrown exception
+     * @return true if exception is thrown for unsupported operation
+     */
+    private static boolean isUnsupportedOperation(Exception exception) {
+        if (exception instanceof SQLFeatureNotSupportedException) {
+            return true;
+        } else if (exception instanceof SQLException sqlException) {
+            return OPERATION_NOT_SUPPORTED.equals(sqlException.getMessage());
+        }
+        return false;
+    }
+
+    /**
+     * Generates savepoint name from the given transaction name.
+     * Some connections don't support '.' (Oracle) and some accept max length 32
+     * so this method will return 'SavePoint' plus transaction name hash code in the result.
+     *
+     * @param transactionName The transaction name
+     * @return savepoint name for given transaction name
+     */
+    private static String getSavepointName(String transactionName) {
```

Comment:
The issue was when transaction name is like `SomeService.someMethod` and Oracle does not accept '.' in save point names. On the other hand, MSSQL limits savepoint name to 32 characters so this is what could work for both cases - return `'SavePoint' + transactionName.hashCode()`. Trimming to 32 might cause some duplicated savepoint names from different methods/transactions so I think this is safer, unless we have to have human readable savepoint name for debug purposes.

Reply:
There's also (possibly?) the option of not using a save point name at all:
https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setSavepoint--

Reply:
I have changed to set savepoint without passing name and then don't need to deal with special chars and max length.